### PR TITLE
[DataComp] Add download images component

### PR DESCRIPTION
**Pull request by NielsRogge**
This PR:

splits up the datacomp folder into 2 pipelines:

- a simple pipeline, just consisting of 3 components, serving as a simple baseline which could serve as a first submission
- a more advanced pipeline, which involves downloading images (using the reusable `download_images` component), and later on also text detection and text recognition 

and improves the `download_images` component to leverage Dask's `map_partitions`.

### CODE REVIEW

File: components/download_images/src/main.py
Comment (RobbeSneyders) [inline components/download_images/src/main.py:161]: Is there still a reason to use a `DaskTransformComponent` for this component? If we use the `PandasTransformComponent`, we get the `map_partitions` for free.

File: examples/pipelines/datacomp/components/filter_text_complexity/src/main.py
Comment (RobbeSneyders) [inline examples/pipelines/datacomp/components/filter_text_complexity/src/main.py:31]: Would remove this function if it's not used. We have it in git, so can always re-add it when needed.

File: examples/pipelines/datacomp/components/load_from_hf_hub/fondant_component.yaml
Comment (RobbeSneyders) [inline examples/pipelines/datacomp/components/load_from_hf_hub/fondant_component.yaml:30]: Shouldn't this be used as the index? Is this missing functionality in the current `read_from_hub` component?

**Review by RobbeSneyders — COMMENTED** [review]
Thanks @NielsRogge!

Did you notice any speedup by using `map_partitions` + `apply` instead of just `apply`?

Don't forget to revert all your local changes before merging.

File: examples/pipelines/datacomp/components/load_from_hf_hub/fondant_component.yaml
Comment (NielsRogge) [reply examples/pipelines/datacomp/components/load_from_hf_hub/fondant_component.yaml:30]: Currently the `load_from_hf_hub` creates a monotonically increasing index: https://github.com/ml6team/fondant/blob/92942afa0d43e150ce77b15f7cc4039d75f222f1/components/load_from_hf_hub/src/main.py#L62-L66.

@PhilippeMoussalli 's PR will remove that and set it in Fondant: https://github.com/ml6team/fondant/pull/345

Comment (RobbeSneyders) [reply examples/pipelines/datacomp/components/load_from_hf_hub/fondant_component.yaml:30]: Yes, but the `uid` is a field from the initial huggingface dataset we're loading? Then I would assume that ideally, we use that column as the index in Fondant?

Comment (NielsRogge) [reply examples/pipelines/datacomp/components/load_from_hf_hub/fondant_component.yaml:30]: Ok but there's no way to specify that currently, right? Would we need to add an `index_column` argument to the `load_from_hf_hub` component?

Comment (RobbeSneyders) [reply examples/pipelines/datacomp/components/load_from_hf_hub/fondant_component.yaml:30]: Yes, that's what I was steering towards. No need to add it in this PR, but important to know that that would be useful.

@PhilippeMoussalli related to my comment on #345

**Comment by RobbeSneyders** [comment]
@NielsRogge LGTM if you can revert all the changes that should not be in git (the commit hashes etc.)

File: examples/pipelines/datacomp/components/load_from_hf_hub/fondant_component.yaml
Comment (RobbeSneyders) [inline examples/pipelines/datacomp/components/load_from_hf_hub/fondant_component.yaml:3]:
```suggestion
image: ghcr.io/ml6team/load_from_hf_hub:dev
```

**Review by RobbeSneyders — APPROVED** [review]

**Pull request merged**
